lib/commands/moveToDelayed-3.lua

```diff
@@ -8,25 +8,28 @@
 
     ARGV[1] delayedTimestamp
     ARGV[2] the id of the job
-    ARGV[3] timestamp
+    ARGV[3] queue token
 
   Output:
     0 - OK
-    -1 - Missing job.
+   -1 - Missing job.
+   -2 - Job is locked.
 
   Events:
     - delayed key.
 ]]
 if redis.call("EXISTS", KEYS[3]) == 1 then
-  local score = tonumber(ARGV[1])
-  if score ~= 0 then
+  local lockKey = KEYS[3] .. ':lock'
+  local lock = redis.call("GET", lockKey)
+  if (not lock) or (lock == ARGV[3]) then 
```

Comment:
`not lock` should be removed since another worker might have already completed the job and deleted the lock. When I was trying to fix this locally, I realized that a bunch of unit tests broke when enforcing that a lock exists in this script. I fixed this by having `Job.prototype.moveToDelayed` take an `ignoreLock` parameter that passed in `0` as the token to the script, and checked it inside the script using:

```
  if ARGV[4] ~= "0" then
    local lockKey = KEYS[3] .. ':lock'
    if redis.call("GET", lockKey) ~= ARGV[4] then
      return -2
    end
  end
```

(this is how it's done in moveToFinished). That way the unit tests could use the special moveToDelayed() function that can ignore the lock.